app/controllers/actions/DataRetrievalAction.scala

```diff
@@ -281,7 +281,14 @@ class RacdacDataRetrievalImpl(
          .getSchemeDetails(psaId, schemeIdType = "srn", srn, Some(true))
          .flatMap(ua => addMinimalFlagsAndUpdateRepository(srn, ua.json, psaId, viewConnector.upsert(request.externalId, _)).map(Some(_)))
 
-       case Some(value) => Future.successful(Some(value))
+       case Some(jsonValue) => (jsonValue \ "srn").validate[String] match {
```

Comment:
This can be simplified. As Jignesh no longer at HMRC, I will fix myself and push up change.

Reply:
done